xmodule/tasks.py

```diff
@@ -0,0 +1,317 @@
+    # might cause conflicts when the same filename is used across imported blocks.
+    if isinstance(source_key, LibraryUsageLocatorV2):
+        all_assets = library_api.get_library_block_static_asset_files(source_key)
+    else:
+        all_assets = []
+
+    for field_name, field in source_block.fields.items():
+        if field.scope not in (Scope.settings, Scope.content):
+            continue  # Only copy authored field data
+        if field.is_set_on(source_block) or field.is_set_on(new_block):
+            field_value = getattr(source_block, field_name)
+            if isinstance(field_value, str):
+                # If string field (which may also be JSON/XML data), rewrite /static/... URLs to point to blockstore
+                for asset in all_assets:
+                    field_value = field_value.replace(f'/static/{asset.path}', asset.url)
+                    # Make sure the URL is one that will work from the user's browser when using the docker devstack
+                    field_value = blockstore_api.force_browser_url(field_value)
+            setattr(new_block, field_name, field_value)
+    new_block.save()
+    store.update_item(new_block, user_id)
+
+    if new_block.has_children:
+        # Delete existing children in the new block, which can be reimported again if they still exist in the
+        # source library
+        for existing_child_key in new_block.children:
+            store.delete_item(existing_child_key, user_id)
+        # Now import the children
+        for child in source_block.get_children():
+            _import_block(store, user_id, child, new_block_key)
+
+    return new_block_key
+
+
+def _filter_child(store, usage_key, capa_type):
+    """
+    Filters children by CAPA problem type, if configured
+    """
+    if usage_key.block_type != "problem":
+        return False
+
+    descriptor = store.get_item(usage_key, depth=0)
+    assert isinstance(descriptor, ProblemBlock)
+    return capa_type in descriptor.problem_types
+
+
+def _get_library(store, library_key, is_v2_lib):
+    """
+    Helper method to get either V1 or V2 library.
+
+    Given a library key like "library-v1:ProblemX+PR0B" (V1) or "lib:RG:rg-1" (v2), return the 'library'.
+
+    is_v2_lib (bool) indicates which library storage should be requested:
+        True - blockstore (V2 library);
+        False - modulestore (V1 library).
+
+    Returns None on error.
+    """
+    if is_v2_lib:
+        try:
+            return library_api.get_library(library_key)
+        except ContentLibrary.DoesNotExist:
+            return None
+    else:
+        try:
+            return store.get_library(
+                library_key, remove_version=False, remove_branch=False, head_validation=False
+            )
+        except ItemNotFoundError:
+            return None
+
+
+def _problem_type_filter(store, library, capa_type):
+    """ Filters library children by capa type"""
```

Comment:
`def _filter_child(store, usage_key, capa_type): -> "Filters children by CAPA problem type, if configured"`

`def _problem_type_filter(store, library, capa_type): -> "Filters library children by capa type "`

Are these redundant or is there a difference?

Reply:
https://github.com/openedx/edx-platform/pull/33263#pullrequestreview-1627813331